File: Lab7/Question9/src/Airplane.java

public class Airplane implements Flyer {
	private String name;

	public Airplane(String name) {
		this.name = name;
	}
	
	@Override
	public boolean canFly() {
		return true;
	}
	
	@Override
	public String getName() {
		return name;
	}
	
	@Override
	public boolean isDangerous() {
		return false;
	}
	
	public static void testAirplane() {
		Airplane frank = new Airplane("Cathay Pacific");
		System.out.println(frank.canFly() == true);
		System.out.println(frank.getName() == "Cathay Pacific");
		System.out.println(frank.isDangerous() == false);
		Flyer ng = new Airplane("Shuyang");
		System.out.println(ng.canFly() == true);
		System.out.println(ng.isDangerous() == false);
		System.out.println(ng.getName() == "Shuyang");
	}
}

File: Lab7/Airplane.kt
class Airplane : Flyer {
    private val name: String

    constructor(name: String) {
        this.name = name
    }

    override fun canFly(): Boolean {
        return true
    }

    override fun getName(): String {
        return name
    }

    override fun isDangerous(): Boolean {
        return false
    }

    companion object {
        fun testAirplane() {
            val frank = Airplane("CathayPacific")
            println(frank.canFly() == true)
            println(frank.getName() == "CathayPacific")
            println(frank.isDangerous() == false)
            val ng: Flyer = Airplane("Shuyang")
            println(ng.canFly() == true)
            println(ng.getName() == "Shuyang")
            println(ng.isDangerous() == false)
        }
    }
}